manifests/service.pp

```diff
@@ -16,19 +16,24 @@
 class nginx::service(
   $service_restart = $::nginx::service_restart,
   $service_ensure  = $::nginx::service_ensure,
+  $service_enable  = $::nginx::service_enable,
   $service_name    = $::nginx::service_name,
   $service_flags   = $::nginx::service_flags,
   $service_manage  = $::nginx::service_manage,
 ) {
 
   assert_private()
 
-  $service_enable = $service_ensure ? {
-    'running' => true,
-    'absent'  => false,
-    'stopped' => false,
-    'undef'   => undef,
-    default   => true,
+  if $service_enable != undef {
+    $service_enable_real = $service_enable
```

Comment:
I think `foo_real` is less encouraged than `_foo`. That said, for `service_enable`, wouldn't it be better to just make the `contain` of the service class conditional? (in https://github.com/voxpupuli/puppet-nginx/blob/master/manifests/init.pp#L176)